ctsit/nacculator
# Telephone Followup Packet version 3.2 Module (C2T / telecognition form)

This change adds the recently released "TFP v3.2" module to nacculator's library. Centers will be able to process data from forms like the T1 and C2T, along with the expanded telehealth regimen in the other forms provided by NACC due to the COVID-19 pandemic.

This gives centers the ability to easily convert their telephone visit data into NACC's format along with regular UDS3 visits and the other common modules.

I also added unit tests for form skipping and some expanded event detection options for variations on the Z1X title within a REDCap project.


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. Ensure that nacculator is installed and configured on your machine, with a complete config file.
2. From the command-line, run: 
$ nacculator_filters nacculator_cfg.ini
3. Then, run:
$ redcap2nacc -tfp <run_(CURRENT-DATE)/final_Update.csv >run_(CURRENT-DATE)/tfp_nacc_complete.txt 2>run_(CURRENT-DATE)/tfp_errors.txt
4. The output should look like regular redcap2nacc output similar to the -ivp and -fvp flags, except with TFP data.
5. If you pass data with errors, the output should look like a message in the error log listing what the problem was.

- Verify the thing does this: runs redcap2nacc with telephone followup packet data
- Verify the thing does not do that: raise false errors, nor pass uncaught errors.


## Affirmations
